desc/objectives/utils.py

```diff
@@ -313,7 +313,7 @@ def factorize_linear_constraints(constraints, objective_args):  # noqa: C901
 
     def project(x):
         """Project a full state vector into the reduced optimization vector."""
-        x_reduced = jnp.dot(Z.T, (x - xp)[unfixed_idx])
```

Comment:
any reason for changing dot to matmul? I think they are basically equivalent for 1-D and 2-D array operations, so I am fine with the change

Reply:
In the dev_guide PR, my math showed `x_reduced` = $Z^T (x - xp)$, so I was expecting to see either `Z.T @ ((x-xp)[unfixed_idx])` or `jnp.matmul(Z.t, (x-xp)[unfixed_idx])` in the code. `jnp.dot` was confusing with the math notation I am familiar with for vectors since $Z^T \cdot (x-xp) = \langle Z^T, x-xp \rangle = Z (x-xp) \neq$ `x_reduced`.
I know now that `dot` is general in numpy to make the operation well-defined, but I liked matmul more